rest/utilities_testing.go

```diff
@@ -32,7 +32,7 @@ import (
 
 var gBucketCounter = 0
 
-type RestTester struct {
+type RestTesterConfig struct {
```

Comment:
I think this approach makes sense, but only for config-style properties.  That looks like SyncFn, DatabaseConfig at minimum - could review usages to see if there's anything else.  

If there's a use case to initialize a RestTester for an existing bucket or server context, I think a separate constructor makes more sense.